bokehjs/test/unit/models/scales/linear_interpolation_scale.ts

```diff
@@ -0,0 +1,23 @@
+import {expect} from "assertions"
+
+import {LinearInterpolationScale} from "@bokehjs/models/scales/linear_interpolation_scale"
+import {Range1d} from "@bokehjs/models/ranges/range1d"
+
+describe("models/scales/linear_interpolation_scale module", () => {
+
+  describe("LinearInterpolationScale model", () => {
+
+    it("should support vector mapping values", () => {
+      const scale = new LinearInterpolationScale({
+        binning: [0, 1, 2, 300, 39999],
+        source_range: new Range1d({start: 0, end: 40000}),
```

Comment:
@philippjfr, @jlstevens, so the root cause of the problem was that `binning[-1] != source_range.end` (in the notebook; this test is a simplification of that). Should binning bounds be equal to range bounds? If that's the case, then this PR doesn't fix anything, and it's binning computation that needs to be addressed.

Reply:
Hmm, I'd have to dig into that again. If I understand correctly `source_range.end` is the upper bound of the colorbar? I don't think they need to be equal although they usually would be.

Reply:
> If I understand correctly source_range.end is the upper bound of the colorbar?

Yes.

Reply:
This PR can be merged regardless, as it makes sure that whatever happens, there won't be out of bounds indexing, but I want to make sure, that if there's a deeper problem with this code, it's actually addressed.

Reply:
Agreed: it is worth investigating if there is a deeper issue but I think protecting against out-of-bounds indexing makes sense regardless.